docs/proposed/conflict-api.rst

```diff
@@ -11,42 +11,77 @@ This document is aimed at programmers working on magic-folder. It is a proposed
 Motivation
 ==========
 
-It is desirable to have an explicit API for dealing with conflicts.
-This allows UI programs to be sure they're accurately indicating intent to magic-folders.
+Under normal operation, magic-folders scans other participants' Snapshots and reflects those changes locally; sometimes, these changes can conflict.
+
+It is desirable to have an explicit API for noticing and resolving conflicts.
+The core of magic-folders operation with users is the filesystem.
+This makes the filesystem also an API and conflicts must be part of that API.
+The command-line interface shall have sub-commands for listing and resolving conficts; these commands will use the HTTP APIs.
+
+
+Filesystem API for Conflicts and Resolution
+===========================================
+
+When a Snapshot is found to conflict with a particluar local file (say ``foo``) a "conflict file" is written beside it, reflecting the other participants' content (``foo.conflict-laptop``).
+That is, ``foo.conflict-laptop`` indicates that the participant "``laptop``" has a conflicting update
+The file ``foo.conflict-laptop`` will contain the downloaded ``.content`` of "``laptop``"'s Snapshot.
+The content of ``foo`` remains what it was when the conflict was detected.
+(Note that when multiple participants exist it's possible to have multiple ``*.confict-*`` files pertaining to a single local file).
+
+List Conflicts
+~~~~~~~~~~~~~~
+
+One can use normal directory-browsing tools such as ``ls`` to notice conflict files.
+
+
+Resolve a Conflict
+~~~~~~~~~~~~~~~~~~
+
+When _all_ ``<relpath>.confict-*`` files for a given root are deleted, the conflict is deemed resolved.
+The resolution is whatever the contents of ``<relpath>`` are currently.
+
+So, to resolve a conflict as "take theirs", one could run: ``mv foo.conflict-laptop foo`` if there was a single conflict from participant "``laptop``".
+
+To resolve a conflict as "take mine", one simply deletes ``foo.conflict-laptop`` if there was a single conflict from participant "``laptop``".
+
 
 
 HTTP API for Conflicts and Resolution
 =====================================
 
 
-
-``GET /v1/conflicts/<folder-name>``
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+``GET /v1/magic-folder/<folder-name>/conflicts``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 Returns a list (possibly empty) of local filesystem paths corresponding to each Snapshot that is currently in a Conflict state in the given magic-folder.
 
 Our content is in the path itself.
-The conflicting "other" content is in ``<path>.conflict-<name>`` where ``<name>`` is the petname of the user who is provided the conflicted content.
+The conflicting "other" content is in ``<path>.conflict-<name>`` where ``<name>`` is the petname of the participant who is provided the conflicted content.
```

Comment:
I think it probably makes sense to include the list of conflict files explicitly too.

Reply:
Why? The file-names are well-defined, right?

Reply:
Well, we'd need to provide a list of participant names we conflict with at least, for them to find the filenames. (Though I'd just as soon include both)

Otherwise, they'd need to list the directory to find all the `.conflict-*` files though. If they need to that anyway, what value does this provide over just listing the folder and noticing the `.conflict-*` files.

Reply:
Yes, the spec says to include a list of the (conflicting) participant-names.